secret-rotator/tests/mockclient.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+// package client implements testing clients, mocked clients, and fixtures utilities.
+// Should be used with caution. Only for testing purpose.
+
+import (
+	"fmt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"strconv"
+	"time"
+)
+
+type MockClient struct {
+	Secrets map[string]map[string]*Secret
```

Comment:
What is this a map of? Guessing project -> secret name -> secret?

Reply:
correct. Will add comments to these.